src/redux/ducks/notifier/watchers.js

```diff
@@ -0,0 +1,62 @@
+import { notify } from './notifier';
+import LOCNoticeModel, {ADDED, REMOVED, UPDATED} from '../../../models/notices/LOCNoticeModel';
+import LOCsManagerDAO from '../../../dao/LOCsManagerDAO';
+import LOCDAO, {Setting} from '../../../dao/LOCDAO';
+
+const watchNewLOCNotify = (account: string) => (dispatch) => LOCsManagerDAO.contract.then(LOCsDeployed => {
+    LOCsManagerDAO._watch(LOCsDeployed.newLOC, (r, block, time, isOld) => {
+        const loc = new LOCDAO(r.args._LOC);
+        loc.loadLOC(account).then(locModel => dispatch(
+            notify(new LOCNoticeModel({time, loc: locModel, action: ADDED}), isOld)
```

Comment:
All watch callbacks should be covered by tests, so you should extract them out of there and write these tests.
Also you should not extract DAO logic out from DAO - don't use contracts here and don't pass result straightly - pack it in your models inside your DAO! '_' means that method is protected or private!!! And you forgot about AbstractContractDAO::_watch id param and to wrap your action-creators code with isOld check.
Please use 'settings/cbe.js' as an example for reworking! Solution for this issues is already provided - you only need to properly implement it! If you have any adequate offers about provided solutions - they should be discussed and perhaps (depending on results of discussions) implemented for all application modules.
(**btw we already discussed it with you! Please be more careful and write down all such moments, so you won't forget it in future!**)